cpp/style.rst

```diff
@@ -195,17 +195,17 @@ The capitalization rule for class names should be all words in the name capitali
 
 .. _style-guide-cpp-3-2:
 
-3-2. Variable names MUST be in mixed case starting with lower case.
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+3-2. Variable names MUST be camelCase or snake_case.
```

Comment:
I guess we don't need to explicitly say "start with lower case"?

Reply:
I don't think it's likely that someone would incorrectly guess that starting with caps is okay, given the examples here and throughout the codebase, and I think the reference to the Python style guide and PEP8 provides clarification for anyone looking hard for a clear ruling.